src/magic_folder/cli.py

```diff
@@ -1112,51 +1143,19 @@ def do_magic_folder(options):
     so.stderr = options.stderr
     f = subDispatch[options.subCommand]
     try:
-        return maybeDeferred(f, so)
+        yield maybeDeferred(f, so)
     except Exception as e:
         print(u"Error: {}".format(e), file=options.stderr)
         if options['debug']:
-            raise
-
-
-class _MagicFolderService(Service):
-    def __init__(self, options):
-        self.options = options
-
-    def startService(self):
-        d = maybeDeferred(do_magic_folder, self.options)
-        d.addBoth(_stop)
-
-def _stop(reason):
-    if isinstance(reason, Failure):
-        print(reason.getTraceback())
-    from twisted.internet import reactor
-    reactor.callWhenRunning(reactor.stop)
-
-# Provide the option parsing helper for the IServiceMaker plugin that lets us
-# have "twist magic_folder ...".
-Options = MagicFolderCommand
-
-# Provide the IService-building helper for the IServiceMaker plugin.
-def makeService(options):
-    """
-    :param MagicFolderCommand options: The parsed options for the comand
-        invocation.
-
-    :return IService: An object providing ``IService`` which performs the
-        magic-folder operation requested by ``options`` when it is started.
-    """
-    service = MultiService()
-    _MagicFolderService(options).setServiceParent(service)
-    if options["coverage"]:
-        coverage_service().setServiceParent(service)
-    return service
-
+            traceback.print_exc(file=options.stderr)
+        raise SystemExit(1)
 
+@inlineCallbacks
 def main(reactor):
     options = MagicFolderCommand()
     options.parseOptions(sys.argv[1:])
-    return do_magic_folder(options)
+    yield do_magic_folder(options)
```

Comment:
not sure I understand? it's a yield because this is a `@inlineCallbacks` method and `do_magic_folder` is also async. I suppose they could be collapsed into one method now, if that's what you mean?

Reply:
There's only one Deferred though right? (going from memory, github is making it incredibly hard for me to see the whole function in this context)  so `return do_magic_folder(options)` without `inlineCallbacks` was fine.